lib/spack/spack/stage.py

```diff
@@ -735,6 +739,188 @@ def cache_local(self):
         tty.msg("Sources for DIY stages are not cached")
 
 
+class CargoStage(object):
```

Comment:
This doesn't just get applied for those packages that derive `CargoPackage`, but any package that uses the `cargo_manifest` directive. This is to support caching of dependencies for packages that use Cargo internally, but where Cargo is not the top-level package.